src/fmeval/eval_algorithms/util.py

```diff
@@ -394,3 +410,148 @@ def get_bert_score(
+        input_key_to_response_keys={DatasetColumns.PROMPT.value.name: [(DatasetColumns.MODEL_OUTPUT.value.name,)]},
+        model_runner=model,
+    )
+    return TransformPipeline([gen_prompt, get_model_response])
+
+
+def compute_and_aggregate_metrics(
+    pipeline: TransformPipeline,
+    dataset: Dataset,
+    dataset_name: str,
+    dataset_prompt_template: Optional[str],
+    eval_name: str,
+    metric_names: List[str],
+    eval_results_path: str,
+    agg_method=MEAN,
+    save: bool = False,
+) -> EvalOutput:
+    """Execute an evaluation algorithm's pipeline on a single dataset to compute algorithm-specific metrics.
+
+    This function is a helper function for the `evaluate` util function below.
+
+
+    :param pipeline: The evaluation algorithm's pipeline, to be executed on one or more datasets.
+    :param dataset: The dataset to execute the algorithm's pipeline on.
+    :param dataset_name: The name of the dataset.
+    :param dataset_prompt_template: The prompt template used to generate the prompts that are fed
+        to the model when it is invoked. Note that this argument is passed purely to be included
+        as metadata in the returned EvalOutput. The `pipeline` argument should already contain
+        the relevant prompt-generation and model-invocation transforms whose behavior is
+        configured by this argument.
+    :param eval_name: The name of the evaluation algorithm.
+    :param metric_names: The names of the metrics that this evaluation algorithm computes.
+    :param eval_results_path: A file containing evaluation results will be stored under this directory.
+    :param agg_method: The aggregation method to use when aggregating the computed metric values.
+        Currently, only MEAN is supported.
+    :param save: If set to true, prompt responses and scores will be saved to a file.
+        The path that this file is stored at is configured by `eval_results_path`.
+
+    :return: A list of EvalOutput objects.
+    """
+    with (timed_block(f"Computing score and aggregation on dataset {dataset_name}", logger)):
+        dataset = pipeline.execute(dataset)
+        dataset_scores, category_scores = aggregate_evaluation_scores(dataset, metric_names, agg_method=agg_method)
+
+        output_path = generate_output_dataset_path(
+            path_to_parent_dir=eval_results_path,
+            eval_name=eval_name,
+            dataset_name=dataset_name,
+        )
+        eval_output = EvalOutput(
+            eval_name=eval_name,
+            dataset_name=dataset_name,
+            prompt_template=dataset_prompt_template,
+            dataset_scores=dataset_scores,
+            category_scores=category_scores,
+            output_path=output_path,
+        )
+
+        if save:  # pragma: no branch
+            save_dataset(
+                dataset=dataset,
+                score_names=metric_names,
+                path=output_path,
+            )
+
+        return eval_output
+
+
+def evaluate(
```

Comment:
Question, wouldn't it be possible/desirable to have this part as a standard implementation of EvalAlgorithmInterface.evaluate? 

In any case I think it'd be better to change the name not to clash with the evaluate. Something like `compute_evaluation` could go

Reply:
Updated the name to `run_evaluation`